misc/runtimes.md

```diff
@@ -0,0 +1,61 @@
+Design rationale:
+
+There are a few alternative HPX/CUDA implementation options:
+
+1) The approach taken here inserts an event just after a CUBLAS function is called and uses it to
+   track when the function has finished. Each task calls `hpx::util::yield_while()` and
+   `cudaEventQuery()` to infrom HPX's scheduler when it's is ready.
```

Comment:
```suggestion
   `cudaEventQuery()` to infrom HPX's scheduler when it is ready.
```

Reply:
I'll fix typos in this document later on. Do we still want to document all the potential approaches, or should I update this to describe only what is currently done?

Reply:
I don't have a strong preference either way. Perhaps you can elaborate on the approach we chose and just mention the rest as bullet-points without going into details?